jolt-core/src/dag/state_manager.rs

```diff
@@ -0,0 +1,195 @@
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::ops::Index;
+use std::sync::{Arc, Mutex};
+
+use crate::field::JoltField;
+use crate::poly::commitment::commitment_scheme::CommitmentScheme;
+use crate::poly::multilinear_polynomial::MultilinearPolynomial;
+use crate::poly::opening_proof::{ProverOpeningAccumulator, VerifierOpeningAccumulator};
+use crate::r1cs::builder::CombinedUniformBuilder;
+use crate::r1cs::inputs::JoltR1CSInputs;
+use crate::r1cs::key::UniformSpartanKey;
+use crate::r1cs::spartan::UniformSpartanProof;
+use crate::subprotocols::sumcheck::SumcheckInstanceProof;
+use crate::utils::transcript::Transcript;
+
+pub type Endianness = bool;
+pub const BIG_ENDIAN: Endianness = false;
+pub const LITTLE_ENDIAN: Endianness = true;
+
+#[derive(Clone, Debug)]
+pub struct OpeningPoint<const E: Endianness, F: JoltField> {
+    pub r: Vec<F>,
+}
+
+impl<const E: Endianness, F: JoltField> OpeningPoint<E, F> {
+    pub fn new(r: Vec<F>) -> Self {
+        Self { r }
+    }
+
+    pub fn endianness(&self) -> &'static str {
+        if E == BIG_ENDIAN {
+            "big"
+        } else {
+            "little"
+        }
+    }
+
+    pub fn match_endianness<const SWAPPED_E: Endianness>(&self) -> OpeningPoint<SWAPPED_E, F>
+    where
+        F: Clone,
+    {
+        let mut reversed = self.r.clone();
+        if E != SWAPPED_E {
+            reversed.reverse();
+        }
+        OpeningPoint::<SWAPPED_E, F>::new(reversed)
+    }
+}
+
+impl<F: JoltField> From<Vec<F>> for OpeningPoint<LITTLE_ENDIAN, F> {
+    fn from(r: Vec<F>) -> Self {
+        Self::new(r)
+    }
+}
+
+impl<F: JoltField> From<Vec<F>> for OpeningPoint<BIG_ENDIAN, F> {
+    fn from(r: Vec<F>) -> Self {
+        Self::new(r)
+    }
+}
+
+impl<F: JoltField> Index<JoltR1CSInputs> for Openings<F> {
+    type Output = Option<F>;
+
+    fn index(&self, index: JoltR1CSInputs) -> &Self::Output {
+        &self[&OpeningsKeys::SpartanZ(index)].1
+    }
+}
+
+#[derive(Hash, PartialEq, Eq, Copy, Clone, Debug)]
+pub enum OpeningsKeys {
+    SpartanZ(JoltR1CSInputs),
+    InstructionTypeFlag(usize),
+    InstructionRa(usize),
+    OuterSumcheckClaims, // (Az, Bz, Cz)
```

Comment:
Does this one contain the RLC of the Az Bz Cz for the inner sumcheck? Which one contains the Az, Bz, Cz of the outer sumcheck?